model_api/python/model_api/tilers/instance_segmentation.py

```diff
@@ -114,28 +111,30 @@ def _merge_results(self, results, shape):
         if np.prod(detections_array.shape):
             detections_array, keep_idxs = multiclass_nms(
                 detections_array,
-                max_num=self.max_pred_number,
-                iou_threshold=self.iou_threshold,
+                max_num=self.max_pred_number,  # type: ignore[attr-defined]
+                iou_threshold=self.iou_threshold,  # type: ignore[attr-defined]
             )
         masks = [masks[keep_idx] for keep_idx in keep_idxs]
 
         merged_vector = np.mean(feature_vectors, axis=0) if feature_vectors else np.ndarray(0)
         saliency_map = self._merge_saliency_maps(saliency_maps, shape, tiles_coords) if saliency_maps else []
 
-        detected_objects = []
-        for i in range(detections_array.shape[0]):
-            label = int(detections_array[i][0])
-            score = float(detections_array[i][1])
-            bbox = list(detections_array[i][2:].astype(np.int32))
-            masks[i] = _segm_postprocess(np.array(bbox), masks[i], *shape[:-1])
-            detected_objects.append(
-                SegmentedObject(*bbox, score, label, self.model.labels[label], masks[i]),
-            )
+        labels, scores, bboxes = np.hsplit(detections_array, [1, 2])
+        labels = labels.astype(np.int32)
+        resized_masks, label_names = [], []
+        for mask, box, label_idx in zip(masks, bboxes, labels):
+            label_names.append(self.model.labels[int(label_idx)])
```

Comment:
Is this conversion caused by np int type? I see astype(np.int32) several lines before, so here can't be a float

Reply:
yeah, it's due to it being a np.array, this is the line before: 
```
labels, scores, bboxes = np.hsplit(detections_array, [1, 2])
label_idx.shape -> (1, )
```